apps/web/src/lib/data/store.ts

```diff
@@ -67,3 +72,35 @@ export const bookReaderKeybindMap$ = writableSubject<BookReaderKeybindMap>({
 const db = browser ? createBooksDb() : import('fake-indexeddb/auto.js').then(() => createBooksDb());
 
 export const database = new DatabaseService(db);
+
+/* eslint-disable no-param-reassign */
+function popoverStore() {
+  const { subscribe, set, update } = writable([]);
+
+  return {
+    subscribe,
+    set,
+    update,
+    add(instance: never) {
+      this.update((instances) => {
+        instances.push(instance);
+        return instances;
+      });
+    },
+    replace(instance: never) {
+      this.update((instances) => {
+        instances = [instance];
+        return instances;
+      });
+    },
+    remove(instance: string) {
+      this.update((instances) => {
+        instances = instances.filter((item) => item !== instance);
+        return instances;
+      });
+    }
+  };
+}
+/* eslint-enable no-param-reassign */
+
+export const popovers = popoverStore();
```

Comment:
Think this should be in popover directory
e.g.
`components/popover/popover.svelte`
`components/popover/popovers.ts`